pkgs/top-level/haskell-ng.nix

```diff
@@ -16,7 +17,10 @@ rec {
     ghc722 = callPackage ../development/compilers/ghc/7.2.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc742 = callPackage ../development/compilers/ghc/7.4.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc763 = callPackage ../development/compilers/ghc/7.6.3.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
-    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix { ghc = compiler.ghc742Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
+    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix {
+      ghc = if pkgs.stdenv.isDarwin then compiler.ghc783Binary else compiler.ghc742Binary;
```

Comment:
Yes, using cpphs sounds like it's worth a try! It would be really nice to use the same bootstrap compiler on Linux and Darwin. If it doesn't work out, then so be it, but it would be great to give it a try.